### DISCUSSION TITLE
ChromeOS build

### DISCUSSION THREAD

**Discussion by zhuiks** [ezra-bible-app/ezra-bible-app#245] [Q&A] [answered]
Chromebooks are gaining popularity. And having a web-technologies as the main stack of Ezra Bible App it seems logical to have it available for ChromeOS platform.

Obviously this needs more research. I just want to put it to start a conversation.

**Comment by zhuiks** [comment]
It's possible now to install Android apps and Linux apps on ChromeOS

**Comment by reyespinosa1996** [comment]
.deb should work on ChromeOS, if someone had an old PC or hardware for a VM there is a Chromium OS bases clone named CloudReady https://www.neverware.com/ 

-- 
 Enviado con Tutanota, el correo seguro y sin publicidad: 
 https://tutanota.com


1 de mayo de 2021 12:05 por ***@***.***:

**Comment by tobias-klein** [comment, marked as answer]
Technically the existing apk file should already work on Chrome OS. You can install apps from the Play Store there! I have never tested this, though and I do not have any user feedback yet. But I think somebody has run Ezra Bible App on an Android TV!
But anyway ... in other words I don't think a separate build would be required for ChromeOS!

**Comment by zhuiks** [comment]
Here is a screenshot running on Chromebook:

![Screenshot 2021-05-01 11 34 52](https://user-images.githubusercontent.com/3584119/116789541-fc415380-aa74-11eb-98d6-5557d231c117.png)

It is running pretty smoothly, But is's on Intel Core m3 machine. It would be nice to check it out on a simpler Chromebook.

I have just a few minor comments: 
-  full-screen button works but it doesn't detect if I switch to fullscreen by native means
- when using touch screen buttons are a bit small. It's especially tricky to select a book from the book selection menu with a touch

I'll try to check with deb.

Maybe under installation instructions there should be a dedicated section about ChromeOS

**Comment by tobias-klein** [comment]
Thanks for trying and posting the screenshot! :)
I suppose the sizes of items are comparable to similar screen resolutions on regular laptop/desktop computers. I can imagine that it may be OK with a mouse but difficult with touch only (buttons may need to be bigger for that). I wonder whether it's possible to detect Chrome OS with JavaScript and/or Cordova facilities. Then we could add a special class on the body element and utilize that in the CSS.
Yes, we can consider Chrome OS as supported option on the website. Would you like to submit a pull request?

**Comment by tobias-klein** [comment]
On large resolutions we could even consider to make the book selection menu bigger by default and increase the font size of the links.

**Comment by zhuiks** [comment]
> Yes, we can consider Chrome OS as supported option on the website. Would you like to submit a pull request?

Yes, I'll submit pull request to the website repository

**Reply by tobias-klein** [reply]
Note that I have just slightly [adjusted the website](https://github.com/ezra-bible-app/ezrabibleapp.net/commit/219b5da0bdd49b37c1714f2d2acd42f2c2e4caac). It was a small change (just mentioning Chromebooks in two places now).

**Comment by tobias-klein** [comment]
Regarding installation instructions I recently added a section about Android on the website. I suppose the general approach is the same (Play Store). So it could also be grouped under the same headline: Android & ChromeOS

**Comment by tobias-klein** [comment]
I suppose you must have been running this experiment on ChromeOS based on automatic [ARM translation](https://developer.android.com/topic/arc/device-support#arm-translation), because currently I am actually not including x86 and x86_64 in the [build configuration / abi filters](https://github.com/ezra-bible-app/ezra-bible-app-cordova/blob/master/build-extras.gradle). This is for sizing reasons. If all abi options are included the resulting APK would be significantly bigger. Currently it's between 45 and 65 MB depending on whether you build for release or debugging. With both x86 options included it would probably double in size.

This is what Google recommends:
![image](https://user-images.githubusercontent.com/17739895/116792437-d6da3700-aac0-11eb-9749-eac443ab7b21.png)

**Comment by zhuiks** [comment]
I think it's different with Chromebooks. This is from [Android developer's](https://developer.android.com/topic/arc):
> Chromebooks run the entire Android OS in a container, similar to Docker or LXC...

**Comment by tobias-klein** [comment]
The runtime environment does not matter. The CPU is decisive. If you have an Intel CPU and an ARM/PPC binary a translation will have to happen, no matter what runtime environment you are using.

**Comment by zhuiks** [comment]
Yes, you are right. I didn't read the docs enough. It's translated.

Do you have only ARM64 build?

Maybe because we are mainly doing staff in WebView everything is running pretty smoothly.

**Comment by tobias-klein** [comment]
Currently both ARM options (32 + 64 bit) are included in the APK!

The binary parts that are impacted are SWORD, node-sword-interface, sqlite and nodejs-mobile.
